src/Amadevus.RecordGenerator.Generators/EqualityGenerators.cs

```diff
@@ -0,0 +1,231 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxKind;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Amadevus.RecordGenerator.Generators
+{
+    internal abstract class EqualityPartialGeneratorBase : PartialGeneratorBase
+    {
+        protected const string EqualsMethodName = "Equals";
+        protected const string GetHashCodeMethodName = "GetHashCode";
+
+        protected string[] wellKnownTypes { get; } = new[] { typeof(bool), typeof(byte), typeof(sbyte), typeof(char), typeof(int), typeof(uint), typeof(long), typeof(ulong), typeof(short), typeof(ushort), typeof(string) }
```

Comment:
A property should be CamelCased. Also I'd prefer `ImmutableArray` here because in array you can change the value by accident.
Other thing: the name could be more useful. Currently it's not correct, because there are many more well known types (non-integral) - so I'd suggest something closer to `OperatorEqualityTypes` that's associated with operator equality used on them.

Reply:
Actually, Microsoft conventions do not specify wether private / protected members should be uppercase or lowercase. In my company we always use lowercase names for internal members, but I don't really care as long as it's used consistently.